### ISSUE TITLE
Rename `swap-endpoint` to better name

### ISSUE THREAD

**Issue by AuHau** [ethersphere/bee#3263]
`swap-endpoint` represent Bee's blockchain connectivity and nowadays is not linked anymore only to SWAP protocol, but also postage-stamps, in close future to the storage loterry etc. It is quite confusing naming, so I would suggest renaming it to something like `rpc-endpoint`. It could be done with deprecation, eq. supporting for some time (until another breaking release?) both options with priority over the new name, communicating this to users and eventually removing `swap-endpoint`.

**Comment by umerm-work** [comment]
Since the swap endpoint is primarily used in storage incentives and the exchange of data. Soo, we may name it the following. 

- settlement-endpoint
- data-endpoint
- incentive-endpoint

**Comment by agazso** [comment]
@umerm-work There is an existing terminology in the Ethereum ecosystem for this and it is RPC endpoint. You can read a bit more here, but feel free to do other research as well.

https://ethereum.stackexchange.com/questions/10681/what-are-ipc-and-rpc

This terminology (as usual with terminologies) are not very accurate, because strictly speaking what is specified here is not an endpoint, but rather a URL, under which the actual endpoints can be found. Nevertheless, this is the accepted terminology, so instead of inventing new (also incorrect, but in different ways) names for this, let's just adopt it.

In other words, I also vote for using `rpc-endpoint`.

**Comment by agazso** [comment]
> The terminology should not lean towards a specific blockchain nomenclature, as there is always the possibility that this project will transition to another blockchain. Given this, it makes more sense from my perspective to call it a `blockchain-endpoint`.

Swarm already transitioned to use another blockchain, currently it uses the Gnosis chain and not the Ethereum mainnet. This was possible because the Gnosis chain also builds on the Ethereum ecosystem (as well as Swarm) and therefore it also uses the same terminology.

Moving to a completely different technology would mean more radical changes and that could warrant to change also the name of the config variable to be used to that blockchain. However this is neither planned, nor really feasible in the near future anyhow.

**Comment by AuHau** [comment]
Agree with @agazso on this. Also, "rpc" is not really Ethereum specific as it stands for "remote procedure call" which is a general concept...

**Comment by mrekucci** [comment]
> Agree with @agazso on this. Also, "rpc" is not really Ethereum specific as it stands for "remote procedure call" which is a general concept...

Yes, that's why I find the term `rpc` too general. It basically says nothing and can refer to anything.

**Comment by aloknerurkar** [comment]
@AuHau Yes I think `blockchain-rpc-endpoint` is the correct one. Although it's long. The CLI has an option to have a shorter alias for this, so I guess having a bigger name should not be a problem. If this is in the config, the user would hardly touch the name.

**Comment by umerm-work** [comment]
So if everyone agrees on `blockchain-rpc-endpoint` we can proceed with it?
